dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry/internal/rest/TelemetryContainerFilter.java

```diff
@@ -72,6 +72,11 @@ public TelemetryContainerFilter() {
 
     @Inject
     public TelemetryContainerFilter(final OpenTelemetry openTelemetry) {
+        //Instrumenter is null if OpenTelemetry is not enabled
+        if(openTelemetry.toString().equals(OpenTelemetry.noop().toString())){
+            this.instrumenter = null;
+            return;
+        }
```

Comment:
This works a bit better here than in `TelemetryClientFilter` because the filter methods already check for the `instrumenter` being `null`. However, since we're now expecting `instrumenter` to be `null` much more often, we should check for that before calling `Context.current();`.

We register the `TelemetryContainerFilter` using a service loader, so there's no opportunity to dynamically skip registering it in the first place, so setting `instrumenter` to `null` here makes sense, but we should still find a better way to check if telemetry is disabled.